### PR TITLE
Add development tools

### LINKED ISSUE #16: Add composer
## User Story

As a developer, I want to see an integration with the composer so that use it in my project.

## Acceptance Criteria
-   [x] Added Composer.
-   [x] Added composer development packages: https://github.com/impress-org/givewp/blob/33ff05033710218f1fdd5008ade2f7b246d1c115/composer.json#L12.
- [x] Defined `autoload` with `GiveAddon` namespace in `composer.json`: https://github.com/impress-org/givewp/blob/33ff05033710218f1fdd5008ade2f7b246d1c115/composer.json#L41
- [x] Defied `scripts` in `composer.json`: https://github.com/impress-org/givewp/blob/33ff05033710218f1fdd5008ade2f7b246d1c115/composer.json#L28

### LINKED ISSUE #17: install and setup node packages
## User Story

As a developer, I want to see an integration with [node development tools package](https://github.com/impress-org/givewp/blob/33ff05033710218f1fdd5008ade2f7b246d1c115/package.json#L33)s.

## Acceptance Criteria
-   [x] Installed [dev depedecies](https://github.com/impress-org/givewp/blob/33ff05033710218f1fdd5008ade2f7b246d1c115/package.json#L33) of GiveWP
- [x] Webpack upgrdated to `4.44.1`.
- [x] Config added for `babel`, `stylelint`, `eslint`, `prettier`, `lintstaged`, `editorconfig` and `husky` . [ Check GiveWP root directory ]
- [x] Added necessary [scripts](https://github.com/impress-org/givewp/blob/33ff05033710218f1fdd5008ade2f7b246d1c115/package.json#L7)
- [x] Removed unnecessary packages from the existing setup.

### PR DESCRIPTION
**Pull request by ravinderk**
Resolves #16 #17

## Description

This PR adds development tools. I also formatted the existing code.

## Affects

I upgrade most of tools and removed which react related package which we can add in another issue after discussion.

## Pre-review Checklist
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions
- Run `npm i`
- Run `npm run dev`

For testing, you can add a testing commit to check if tools apply of your commit or not.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,65 +1,80 @@
...
+        "lint:php": "./vendor/bin/phpcs -v --colors --standard=phpcs.ruleset.xml",
+        "lint:php:fix": "./vendor/bin/phpcbf -v --colors --standard=phpcs.ruleset.xml",
+        "lint:textdomain": "node ./wp-textdomain.js",
+        "lint:scss": "stylelint assets/src/**/*.scss"
+    },
+    "repository": {
+        "type": "git",
+        "url": "git+https://github.com/impress-org/give-addon-boilerplate.git"
+    },
+    "author": "WordImpress, LLC",
```
Comment (alaca) [inline package.json]: The author should be Impress instead of WordImpress.

File: give-addon-boilerplate.php [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?php
 /**
  * Plugin Name: Give - Addon Boilerplate
- * Plugin URI:  https://github.com/impress-org/give-addon-boilerplate
+ * Plugin URI:  https://github.com/impress-org/givewp-addon-boilerplate
  * Description: A demo Addon to serve as a boilerplate for devs to better understand how to extend the Give Donation plugin for WordPress.
  * Version:     1.0
  * Author:      WordImpress, LLC
```
Comment (alaca) [inline give-addon-boilerplate.php]: The author should be Impress instead of WordImpress.

**Review by alaca — CHANGES_REQUESTED** [review]

File: give-addon-boilerplate.php (hunk 2 above)
Comment (ravinderk) [reply give-addon-boilerplate.php]: fixed: https://github.com/impress-org/givewp-addon-boilerplate/pull/19/commits/491dfe5b423837a1db08ea50b80ecfe9afe5055e

File: package.json (hunk 1 above)
Comment (ravinderk) [reply package.json]: fixed: https://github.com/impress-org/givewp-addon-boilerplate/pull/19/commits/491dfe5b423837a1db08ea50b80ecfe9afe5055e

**Comment by ravinderk** [comment]
@alaca changes implemented as per your suggestion

**Pull request merged**